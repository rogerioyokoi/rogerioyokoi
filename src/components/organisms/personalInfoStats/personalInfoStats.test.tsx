import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { axe } from 'vitest-axe';
import PersonalInfoStats from './personalInfoStats';

describe('Components > Organisms PersonalInfoStats', () => {
  describe('Snapshot', () => {
    it('Should match snapshot', () => {
      const { container } = render(<PersonalInfoStats />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('Accessibility Tests (A11y)', () => {
    it('Should not have accessibility violations with only title', async () => {
      const { container } = render(<PersonalInfoStats />);

      expect(await axe(container)).toHaveNoViolations();
    });
  });

  describe('Rendering', () => {
    it('Should render all stats correctly', () => {
      const { queryAllByTestId } = render(<PersonalInfoStats />);

      const iElmStatsCard = queryAllByTestId('card-container');

      expect(iElmStatsCard.length).toBe(4);
    });
  });

  describe('CSS Classes', () => {
    it('Should apply correct classes to container', () => {
      const { getByTestId } = render(<PersonalInfoStats />);

      const iElmContainer = getByTestId('personal-info-container');

      expect(iElmContainer).toHaveClass('w-full grid grid-cols-2 gap-6');
    });
  });
});
