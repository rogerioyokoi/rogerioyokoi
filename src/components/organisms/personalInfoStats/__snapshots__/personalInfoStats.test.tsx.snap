// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > Organisms PersonalInfoStats > Snapshot > Should match snapshot 1`] = `
<div>
  <section
    class="w-full grid grid-cols-2 gap-6"
    data-testid="personal-info-container"
  >
    <div
      class="border border-gray-300 dark:border-gray-700 rounded-lg shadow-md"
      data-testid="card-container"
    >
      <header
        class="px-10 pt-6 pb-1"
        data-testid="card-header"
      >
        <h3
          class="relative text-5xl mb-2 text-amber-500 font-bold"
          data-testid="stats-card-header"
        >
          +10
        </h3>
      </header>
      <div
        class="py-6 px-10 px-10 pt-2 pb-6"
        data-testid="card-content"
      >
        <div
          class="relative"
        >
          <div
            class="absolute left-0 top-[44%] h-[4px] w-6 bg-gray-300 dark:bg-gray-700 rounded-[10px]"
            data-testid="stats-card-decoration"
          />
          <p
            class="m-0 ml-8 relative tracking-wide text-sm"
            data-testid="stats-card-children"
          >
            anos de
             
            <span
              class="block font-bold"
            >
              experiÃªncia
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="border border-gray-300 dark:border-gray-700 rounded-lg shadow-md"
      data-testid="card-container"
    >
      <header
        class="px-10 pt-6 pb-1"
        data-testid="card-header"
      >
        <h3
          class="relative text-5xl mb-2 text-amber-500 font-bold"
          data-testid="stats-card-header"
        >
          +6
        </h3>
      </header>
      <div
        class="py-6 px-10 px-10 pt-2 pb-6"
        data-testid="card-content"
      >
        <div
          class="relative"
        >
          <div
            class="absolute left-0 top-[44%] h-[4px] w-6 bg-gray-300 dark:bg-gray-700 rounded-[10px]"
            data-testid="stats-card-decoration"
          />
          <p
            class="m-0 ml-8 relative tracking-wide text-sm"
            data-testid="stats-card-children"
          >
            anos com
             
            <span
              class="block font-bold"
            >
              ReactJS
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="border border-gray-300 dark:border-gray-700 rounded-lg shadow-md"
      data-testid="card-container"
    >
      <header
        class="px-10 pt-6 pb-1"
        data-testid="card-header"
      >
        <h3
          class="relative text-5xl mb-2 text-amber-500 font-bold"
          data-testid="stats-card-header"
        >
          +2
        </h3>
      </header>
      <div
        class="py-6 px-10 px-10 pt-2 pb-6"
        data-testid="card-content"
      >
        <div
          class="relative"
        >
          <div
            class="absolute left-0 top-[44%] h-[4px] w-6 bg-gray-300 dark:bg-gray-700 rounded-[10px]"
            data-testid="stats-card-decoration"
          />
          <p
            class="m-0 ml-8 relative tracking-wide text-sm"
            data-testid="stats-card-children"
          >
            anos com
             
            <span
              class="block font-bold"
            >
              VueJS (2)
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="border border-gray-300 dark:border-gray-700 rounded-lg shadow-md"
      data-testid="card-container"
    >
      <header
        class="px-10 pt-6 pb-1"
        data-testid="card-header"
      >
        <h3
          class="relative text-5xl mb-2 text-amber-500 font-bold"
          data-testid="stats-card-header"
        >
          +2
        </h3>
      </header>
      <div
        class="py-6 px-10 px-10 pt-2 pb-6"
        data-testid="card-content"
      >
        <div
          class="relative"
        >
          <div
            class="absolute left-0 top-[44%] h-[4px] w-6 bg-gray-300 dark:bg-gray-700 rounded-[10px]"
            data-testid="stats-card-decoration"
          />
          <p
            class="m-0 ml-8 relative tracking-wide text-sm"
            data-testid="stats-card-children"
          >
            anos com
             
            <span
              class="block font-bold"
            >
              Angular 2+
            </span>
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
